spynnaker/pyNN/models/common/abstract_neuron_recordable.py

```diff
@@ -93,3 +93,15 @@ def get_neuron_sampling_interval(self, variable):
         :return: Sampling interval in microseconds
         :rtype: float
         """
+
+    @abstractmethod
+    def get_expected_n_rows(
+            self, n_machine_time_steps, sampling_rate, vertex, variable):
+        """ Returns the number of expected rows for a given runtime
+
+        :param n_machine_time_steps: map of vertex to steps.
```

Comment:
Types needed for arguments and for return. It's more important for abstract classes than for concrete ones.

Reply:
fixed